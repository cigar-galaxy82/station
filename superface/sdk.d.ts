export { CommunicationSendSmsProfile } from "./types/communication/send-sms";
export { VcsUserReposProfile } from "./types/vcs/user-repos";
export { VcsPullRequestsProfile } from "./types/vcs/pull-requests";
export { VcsSingleFileContentProfile } from "./types/vcs/single-file-content";
export { AddressCleanAddressProfile } from "./types/address/clean-address";
export { CommunicationSendEmailProfile } from "./types/communication/send-email";
export { StarwarsCharacterInformationProfile } from "./types/starwars/character-information";
export { VcsPullRequestProfile } from "./types/vcs/pull-request";
export { AddressGeocodingProfile } from "./types/address/geocoding";
export { DeliveryTrackingShipmentInfoProfile } from "./types/delivery-tracking/shipment-info";
export declare const SuperfaceClient: new () => import("@superfaceai/one-sdk/dist/client/public/client").TypedSuperfaceClient<{
    "delivery-tracking/shipment-info": {
        ShipmentInfo: [import("./types/delivery-tracking/shipment-info").DeliveryTrackingShipmentInfoShipmentInfoInput, import("./types/delivery-tracking/shipment-info").DeliveryTrackingShipmentInfoShipmentInfoResult];
    };
    "address/geocoding": {
        Geocode: [import("./types/address/geocoding").AddressGeocodingGeocodeInput, import("./types/address/geocoding").AddressGeocodingGeocodeResult];
        ReverseGeocode: [import("./types/address/geocoding").AddressGeocodingReverseGeocodeInput, import("./types/address/geocoding").AddressGeocodingReverseGeocodeResult];
    };
    "vcs/pull-request": {
        PullRequest: [import("./types/vcs/pull-request").VcsPullRequestPullRequestInput, import("./types/vcs/pull-request").VcsPullRequestPullRequestResult];
    };
    "starwars/character-information": {
        RetrieveCharacterInformation: [import("./types/starwars/character-information").StarwarsCharacterInformationRetrieveCharacterInformationInput, import("./types/starwars/character-information").StarwarsCharacterInformationRetrieveCharacterInformationResult];
    };
    "communication/send-email": {
        SendEmail: [import("./types/communication/send-email").CommunicationSendEmailSendEmailInput, import("./types/communication/send-email").CommunicationSendEmailSendEmailResult];
        SendTemplatedEmail: [import("./types/communication/send-email").CommunicationSendEmailSendTemplatedEmailInput, import("./types/communication/send-email").CommunicationSendEmailSendTemplatedEmailResult];
    };
    "address/clean-address": {
        CleanAddress: [import("./types/address/clean-address").AddressCleanAddressCleanAddressInput, import("./types/address/clean-address").AddressCleanAddressCleanAddressResult];
    };
    "vcs/single-file-content": {
        SingleFileContent: [import("./types/vcs/single-file-content").VcsSingleFileContentSingleFileContentInput, import("./types/vcs/single-file-content").VcsSingleFileContentSingleFileContentResult];
    };
    "vcs/pull-requests": {
        PullRequests: [import("./types/vcs/pull-requests").VcsPullRequestsPullRequestsInput, import("./types/vcs/pull-requests").VcsPullRequestsPullRequestsResult];
    };
    "vcs/user-repos": {
        UserRepos: [import("./types/vcs/user-repos").VcsUserReposUserReposInput, import("./types/vcs/user-repos").VcsUserReposUserReposResult];
    };
    "communication/send-sms": {
        SendMessage: [import("./types/communication/send-sms").CommunicationSendSmsSendMessageInput, import("./types/communication/send-sms").CommunicationSendSmsSendMessageResult];
        RetrieveMessageStatus: [import("./types/communication/send-sms").CommunicationSendSmsRetrieveMessageStatusInput, import("./types/communication/send-sms").CommunicationSendSmsRetrieveMessageStatusResult];
    };
}>;
export declare type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
