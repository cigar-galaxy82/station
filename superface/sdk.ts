import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendEmail } from "./types/communication/send-email";
import { communicationSendMessage } from './types/communication/send-message';

export { CommunicationSendEmailProfile } from "./types/communication/send-email";
export const typeDefinitions = {
    ...communicationSendEmail,
    ...communicationSendMessage
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
