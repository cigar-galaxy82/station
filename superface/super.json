{
  "profiles": {
    "communication/send-message": {
      "file": "./grid/communication/send-message/profile.supr",
      "providers": {
        "twilio": {
          "file": "./grid/communication/send-message/maps/twilio.suma",
          "defaults": {
            "SendMessage": {
              "input": {
                "from": "+13345090396",
                "TWILIO_SID": "$TWILIO_USERNAME"
              }
            }
          }
        },
        "tyntec": {
          "file": "./grid/communication/send-message/maps/tyntec.suma",
          "defaults": {
            "SendMessage": {
              "input": {
                "from": "tyntec"
              }
            }
          }
        }
      }
    },
    "vcs/user-repos": {
      "file": "./grid/vcs/user-repos/profile.supr",
      "providers": {
        "github": {
          "file": "./grid/vcs/user-repos/maps/github.suma"
        }
      }
    },
    "vcs/pull-requests": {
      "version": "1.0.0",
      "providers": {
        "gitlab": {
          "file": "./grid/vcs/pull-requests/maps/gitlab.suma"
        }
      }
    }
  },
  "providers": {
    "github": {
      "file": "../providers/github.json",
      "security": []
    },
    "twilio": {
      "file": "../providers/twilio.json",
      "security": [
        {
          "id": "basic",
          "username": "$TWILIO_USERNAME",
          "password": "$TWILIO_PASSWORD"
        }
      ]
    },
    "tyntec": {
      "file": "../providers/tyntec.json",
      "security": [
        {
          "id": "apikey",
          "apikey": "$TYNTEC_API_KEY"
        }
      ]
    },
    "gitlab": {
      "file": "../providers/gitlab.json",
      "security": []
    }
  }
}