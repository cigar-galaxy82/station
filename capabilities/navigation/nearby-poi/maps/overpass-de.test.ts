import { SuperfaceClient } from '../../../../superface/sdk';

describe('navigation/nearby-poi/overpass-de-typed', () => {
  beforeAll(() => {
    jest.setTimeout(15000)
  })
  
  it('finds Astronomy Cafe in Greenwich', async () => {
    const client = new SuperfaceClient();
    const profile = await client.getProfile('navigation/nearby-poi');
    const provider = await client.getProvider('overpass-de');
    const usecase = profile.useCases.NearbyPoi;

    expect(provider).not.toBeUndefined();
    expect(usecase).not.toBeUndefined();

    const result = await usecase.perform(
      {
        center: {
          latitude: 51.477,
          longitude: 0.0
        },
        radius: 100,
        categories: ['CAFE']
      },
      { provider }
    );
    
    const points = result.unwrap();

    expect(
      points.every(
        p => p.categories.includes('CAFE')
      )
    ).toBe(true)

    expect(
      points
    ).toContainEqual(
      {
        coordinates: {
          latitude: 51.4768380,
          longitude: -0.0006877
        },
        name: 'Astronomy Caf√©',
        categories: ['CAFE']
      }
    )
  });
});