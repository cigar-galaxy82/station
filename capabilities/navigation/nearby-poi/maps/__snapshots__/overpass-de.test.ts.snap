// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation/nearby-poi/overpass-de find nearby pois should find astronomy cafe 1`] = `
Ok {
  "value": Array [
    Object {
      "categories": Array [
        "CAFE",
      ],
      "coordinates": Object {
        "latitude": 51.476838,
        "longitude": -0.0006877,
      },
      "name": "Astronomy Caf√©",
    },
  ],
}
`;
